### PR TITLE
Fix test skip handling while test initialization - device_timer

### LINKED ISSUE #664: No test status when TEST_SKIP in case of InitCL
Hi 
I see truncated log from test binary in case of cases for OpenCL 2.X and executed on platform OpenCL 1.X
example:
`test_device_timer.exe
 Initializing random seed to 0.
Requesting Default device based on command line for platform index 0 and device index 0
Compute Device Name = Intel(R) UHD Graphics 605, Compute Device Vendor = Intel(R) Corporation, Compute Device Version = OpenCL 1.2 NEO , CL C Version = OpenCL C 1.2
Supports single precision denormals: YES
sizeof( void*) = 8  (host)
sizeof( void*) = 4  (device)
`
This situation is related with PR #628 

We should print something on output in situation of `return TEST_SKIP;` 
However I am wondering how and what . The problem is that we do not reach calling function https://github.com/KhronosGroup/OpenCL-CTS/blob/master/test_common/harness/testHarness.cpp#L471
where print_results function is called. 
Do you have any suggestion what should be printed in case of 
this skip:
https://github.com/KhronosGroup/OpenCL-CTS/blob/master/test_conformance/device_timer/main.cpp#L41

and this:
https://github.com/KhronosGroup/OpenCL-CTS/blob/master/test_conformance/device_timer/main.cpp#L62

### PR DESCRIPTION
**Pull request by gwawiork**
This is part of the fix #664

### CODE REVIEW

File: test_common/harness/testHarness.cpp [hunk 1]
```diff
@@ -67,6 +67,21 @@ int runTestHarness( int argc, const char *argv[], int testNum, test_definition t
                           ( imageSupportRequired ) ? verifyImageSupport : NULL );
 }
 
+int skip_init_info() {
+    log_info("Test skipped while initialization\n");
+    log_info("PASSED %s.\n", "test");
```
Comment (kpet) [inline test_common/harness/testHarness.cpp]: As per previous comments, I think my preference would be to say `SKIPPED` here with the total number of tests. We should have a clear distinction between passing and skipping tests.

Also, I don't understand why `test` is printed via a format string.

Comment (gwawiork) [reply test_common/harness/testHarness.cpp]: regarding `test` - its a mistake.  Are we interested in printing full list of tests skipped ?

Comment (gwawiork) [reply test_common/harness/testHarness.cpp]: if no list of tests then:

Test skipped (requires at least version 2.1, but the device reports version 1.2)
Test skipped while initialization
SKIPPED 2 of 2 tests.

Comment (kpet) [reply test_common/harness/testHarness.cpp]: I wouldn't try to print the list. This really belongs in the JSON results. The sample output looks good to me.

Comment (gwawiork) [reply test_common/harness/testHarness.cpp]: updated

**Comment by gwawiork** [comment]
Thanks. Then please merge. It allows me go with changes in other tests.

**Comment by kpet** [comment]
I was just giving others a chance to comment. Merging to enable forward progress.

**Pull request merged**